### ISSUE TITLE
LTA error when making a subset out of a coercer

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#2446]
Was wondering if it would work and got a bad error:

```
+Zoffix │ m: subset Foo of Int(); -> Foo {}(42)
+camelia │ rakudo-moar 266af373d: OUTPUT: «===SORRY!===␤Cannot find method 'nominalize' on object of type Perl6::Metamodel::CoercionHOW␤»
```

**Comment by MasterDuke17** [comment]
Fwiw, this patch:
```
diff --git a/src/Perl6/Metamodel/CoercionHOW.nqp b/src/Perl6/Metamodel/CoercionHOW.nqp
index fe6ebb9ff..f98fc24ba 100644
--- a/src/Perl6/Metamodel/CoercionHOW.nqp
+++ b/src/Perl6/Metamodel/CoercionHOW.nqp
@@ -54,6 +54,14 @@ class Perl6::Metamodel::CoercionHOW
         check_instantiated($coercion_type);
         nqp::typeparameterat($coercion_type, 1)
     }
+
+    method nominalize($obj) {
+        my $constraint_type := $obj.HOW.constraint_type($obj);
+        $constraint_type.HOW.archetypes.nominal ??
+            $constraint_type !!
+            $constraint_type.HOW.nominalize($constraint_type)
+    }
 }
 BEGIN {
     my $root := nqp::newtype(Perl6::Metamodel::CoercionHOW, 'Uninstantiable');
```
causes the error to change to:
```
Constraint type check failed in binding to parameter '<anon>'; expected Foo but got Int (42)
  in block <unit> at -e line 1
```

**Comment by vrurg** [comment]
Fixed with #4058
